docs/DeveloperGuide.md

```diff
@@ -1,34 +1,346 @@
+revise the subjects. Furthermore, they can take quizzes and see if they grasp the topics or subjects well
+enough to attend their school based exams.
+
+Students who find it hard to keep track of their assignments would also find this application
+useful since it lets the students add their tasks. The application then reminds the students
+regarding the upcoming tasks, which would prompt the students to prepare them.
+
+Students who find it difficult to organise woudl find this app useful since it
+offers a one stop solution to manage both their revision and their upcoming tasks .
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
+|v1.0|new user| add a flash card intuitively|don't get put off by learning a new program |
+|v1.0|long time user| categorize the flashcards to different categories|I know where to look for which flashcard|
+|v1.0|organised person| i would want a program that looks neat| will want to use the program|
+|v1.0|potential user| default flashcard being shown to me| understand how a flashcard looks in this application|
+|v1.0|language student|type in all my notes|practice the spelling of the words|
+|v1.0|student|revise all my notes for different subjects|do not have to use multiple applications/websites to revise|
+|v1.0|student| to have colour codes for my notes|quickly glance through and know how much I need to study|
+|v1.0|student|to be able to review them at once or print them|use them offline or as cheat sheets|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v2.0|forgetful student|be notified of assignments which are due in a week|prepare for them|
+|v2.0|developer|log messages from the application|troubleshoot for errors|
+
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+The application satisfies the following requirements:
+ * Portability: Users can transfer their data from the application since all the data is stored in a txt file.
+ * Modifiability: Since the code is broken down into different packages and classes, developers would
+ be able to modify the application with ease.
+ * Simplicity: The application comes with a detailed user guide along with the help commands. Hence, users can always 
+ refer to these to understand on how to use the application.
+ * Compatibility: The application runs on various operating systems: Windows, Linux etc
+ * Testability: The application is built using Gradle, a tool that offers various testing capabilities. Furthermore,
+ developers could also modify the input.txt file to provide a set of inputs and test if the application returns the 
+ expected output.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Revised* - The main class which initiates the whole application.
```

Comment:
Perhaps you can link the following items to their respective use case or vice versa (from their use case to glossary). This will help improve the navigability within you DG